nncf/common/pruning/operations.py

```diff
@@ -199,12 +201,16 @@ def generate_output_mask(cls, node: NNCFNode, graph: NNCFGraph,
         input_edges = graph.get_input_edges(node)
         previous_nodes = [edge.from_node for edge in input_edges]
         input_masks = [input_node.data['output_mask'] for input_node in previous_nodes]
+        input_masks = [mask[node.node_name] if isinstance(mask, dict) else mask for mask in input_masks]
 
         not_empty_masks = [mask for mask in input_masks if mask is not None]  # type: List[NNCFTensor]
         if not not_empty_masks:
             return None
 
         first_non_empty_mask = not_empty_masks[0]
+        if isinstance(first_non_empty_mask, dict):
+            first_non_empty_mask = first_non_empty_mask[node.node_name]
+
```

Comment:
This change could be removed completely, please check

Reply:
Fix it. Thank you for letting me know this.